### PR TITLE
Add Google Cloud Text-to-Speech Support

### PR DESCRIPTION
**Pull request by UncIeRick**
This pull request introduces several changes aimed at adding Google Cloud Text-to-Speech (TTS) support to the application:

1. GoogleCloudTTS Implementation: I've implemented a preliminary version of the GoogleCloudTTS class that follows the existing pattern used in the ElevenLabs TTS class. This new class allows the application to use Google Cloud TTS to synthesize speech from text.

2. New Dependencies: The requirements.txt file has been updated to include the dependencies needed for Google Cloud TTS support.

3. TTS Engine Selection: The get_text_to_speech() function has been updated to allow for the selection of the TTS engine via an environment variable. If the TEXT_TO_SPEECH_USE environment variable is set to GOOGLE_TTS, the application will use the GoogleCloudTTS class. If the variable is set to ELEVEN_LABS or not set at all, the application will use the ElevenLabs class.

4. Google Cloud TTS Access Token: To use Google Cloud TTS, you'll need to download a service account key file from the Google Cloud Console. This file contains the credentials the application needs to authenticate with the Google Cloud TTS API. 
https://cloud.google.com/text-to-speech/docs/before-you-begin#create_a_json_key_for_your_service_account

5. Environment Variable Changes: To use the Google Cloud TTS, you'll need to set the TEXT_TO_SPEECH_USE environment variable to GOOGLE_TTS and provide the path to your service account key file.

6. Next Steps: Customized Voices Support: Extend the stream function to support customized voices. This would allow users to send specific voice parameters to the Google Cloud TTS API, further tailoring the synthesized speech to their preferences.

Please review these changes and let me know if you have any questions or feedback.

### CODE REVIEW

File: realtime_ai_character/audio/text_to_speech/google_cloud_tts.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        'voice': {
+            'languageCode': 'en-US',
+            'name': 'en-US-Standard-C',
+            'ssmlGender': 'NEUTRAL'
+        },
+        'audioConfig': {
+            'audioEncoding': 'MP3'
+        }
+    },
+    'service_account_file': 'x-micron-377920-337f8e45a6b3.json'
```
Comment (Shaunwei) [inline realtime_ai_character/audio/text_to_speech/google_cloud_tts.py:32]: what's in service_account_file?

Comment (Shaunwei) [reply realtime_ai_character/audio/text_to_speech/google_cloud_tts.py:32]: this shouldn't be hardcoded. u can check how .env.example file and see how google service account was used

**Comment by UncIeRick** [comment]
Hi Shaun, I've added the GOOGLE_TTS_SERVICE_ACCOUNT_FILE in the .env.example, does it solve the problem?

File: .env.example [hunk 2]
```diff
@@ -29,6 +29,7 @@ OPEN_AI_WHISPER_API_KEY=YOUR_API_KEY
 # Only "ELEVEN_LABS" for now
 TEXT_TO_SPEECH_USE=ELEVEN_LABS
 ELEVEN_LABS_API_KEY=YOUR_API_KEY
+GOOGLE_TTS_SERVICE_ACCOUNT_FILE=YOUR_JSON_FILE
```
Comment (Shaunwei) [inline .env.example:32]: question, can TTS and STT use the same account file? see line 25

Comment (UncIeRick) [reply .env.example:32]: Yes, the same service account key file can be used for both TTS and STT in Google Cloud, but the TTS api needs to be enabled. I will remove the redundant line. 
Regarding the implementation of Google's TTS, should we follow the current httpx approach as it is in elevenlabs or use the texttospeech.TextToSpeechClient library from Google Cloud as it is in Google STT?

Comment (Shaunwei) [reply .env.example]: I think as long as the TTS interface is the same, it's okay to have different implementation inside

**Review by pycui — APPROVED** [review]
Tested and working fine. Thanks!

**Pull request merged**